File: q1710/q1710.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (int[] b1, int[] b2) -> (b2[1] - b1[1]));
        int i = 0;
        int res = 0;
        while (i < boxTypes.length && truckSize > 0) {
            res += Math.min(truckSize, boxTypes[i][0]) * boxTypes[i][1];
            truckSize -= boxTypes[i][0];
            i++;
        }
        return res;
    }
}

File: q1710/q1710.cpp
class Solution
{
public:
    int maximumUnits(vector<vector<int>> &boxTypes, int truckSize)
    {
        sort(boxTypes.begin(), boxTypes.end(), [](auto const &a, auto const &b)
             { return a[1] > b[1]; });
        int res = 0;
        int i = 0;
        int n = boxTypes.size();
        while (i < n && truckSize > 0)
        {
            res += min(boxTypes[i][0], truckSize) * boxTypes[i][1];
            truckSize -= boxTypes[i][0];
            i++;
        }
        return res;
    }
};